### ISSUE TITLE
Small bugs with BOOLEAN in SQL

### ISSUE THREAD

**Issue by ImeevMA** [tarantool/tarantool#4462]
Tarantool version: 2.3.0-77-ga850acf

OS version: Ubuntu 16.04

Bug description: Small bugs with type BOOLEAN.

Steps to reproduce:

1) Result of implicit cast from BOOLEAN to TEXT is not equal to the explicit one:
```
tarantool> SELECT GROUP_CONCAT(true), GROUP_CONCAT(CAST(true AS TEXT));
---
- metadata:
  - name: GROUP_CONCAT(true)
    type: string
  - name: GROUP_CONCAT(CAST(true AS TEXT))
    type: string
  rows:
  - ['true', 'TRUE']
...
```

2) The IN operator throws a "Type Mismatch" error when the right value is the result of a sub-select:
```
tarantool> SELECT true IN (1,2,3);
---
- metadata:
  - name: true IN (1,2,3)
    type: boolean
  rows:
  - [false]
...

tarantool> SELECT true IN (VALUES(1), (2), ('3'));
---
- null
- 'Type mismatch: can not convert UNSIGNED to BOOLEAN'
...
```
3) Function LENGTH() returns NULL when argument is BOOLEAN:
```
tarantool> SELECT length(true), length(false);
---
- metadata:
  - name: length(true)
    type: integer
  - name: length(false)
    type: integer
  rows:
  - [null, null]
...
```

**Comment by Gerold103** [comment]
Actually it is even more interesting:
```
tarantool> box.execute("SELECT true IN (1, 2, 3);")
---
- metadata:
  - name: true IN (1, 2, 3)
    type: boolean
  rows:
  - [false]
...

tarantool> box.execute("SELECT true IN (1, 2);")
---
- null
- 'Type mismatch: can not convert UNSIGNED to BOOLEAN'
...
```

**Comment by Gerold103** [comment]
And more: 'integer' type is reported in lowercase, while 'UNSIGNED' and others in uppercase:
```
tarantool> box.execute("SELECT true IN (VALUES('1'), (2), ('3'));")
---
- null
- 'Type mismatch: can not convert UNSIGNED to BOOLEAN'
...

tarantool> box.execute("SELECT true IN (VALUES('1'), ('2'), ('3'));")
---
- null
- 'Type mismatch: can not convert TEXT to BOOLEAN'
...

tarantool> box.execute("SELECT true IN (VALUES('1'), ('2'), (3));")
---
- null
- 'Type mismatch: can not convert TRUE to integer'
...
```

**Comment by ImeevMA** [comment]
The first issue was fixed by this commit: c2be845858cbb04ea6e70cf8e2b9e928bcc738fb
The third issue was fixed by this commit: 86b0f21d69db9661d4b9f35f735a748373420ae9
The second issue was moved to a different ticket: #4692
